continuous_eval/metrics/code/sql/sql_deterministic_metrics.py

```diff
@@ -0,0 +1,36 @@
+from typing import List, Union
+
+import sqlparse
+
+from continuous_eval.metrics.base import Metric
+
+
+class SQLSyntaxMatch(Metric):
+    """
+    This metric evaluates the syntactic similarity between the generated SQL query and a set of ground truth queries.
+    It uses the sqlparse library to format and compare the SQL queries.
+    """
+
+    def __call__(self, answer: str, ground_truth_answers: Union[List[str], str]):
```

Comment:
This would be better I think: https://github.com/tobymao/sqlglot?tab=readme-ov-file#ast-diff